spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,653 @@
+    :param list: metrics_patients: list metric value of level above, below and at compression of patient.
+    :param list: metrics_HC: list metric value of level above, below and at compression of healthy
+    controls.
+    :return float: MSCC normalized in %
+    """
+    ma = metrics_patients[0]/metrics_HC[0]
+    mb = metrics_patients[1]/metrics_HC[1]
+    mi = metrics_patients[2]/metrics_HC[2]
+    return metric_ratio(ma, mb, mi)
+
+
+def save_csv(fname_out, level, metric, metric_ratio, metric_ratio_norm, filename):
+    """
+    Save .csv file of MSCC results.
+    :param fname_out:
+    :param level: int: Level of compression.
+    :param metric: str: metric to perform normalization
+    :param metric_ratio: float:
+    :param metric_ratio_norm:
+    :param filename: str: input filename
+    :return:
+    """
+    if not os.path.isfile(fname_out):
+        with open(fname_out, 'w') as csvfile:
+            header = ['filename', 'Compression Level', metric + ' ratio', 'Normalized ' + metric + ' ratio']
+            writer = csv.DictWriter(csvfile, fieldnames=header)
+            writer.writeheader()
+    with open(fname_out, 'a') as csvfile:
+        csv_writer = csv.writer(csvfile, delimiter=',')
+        line = [filename, level, metric_ratio, metric_ratio_norm]
+        csv_writer.writerow(line)
+
+
+def main(argv: Sequence[str]):
+    parser = get_parser()
+    arguments = parser.parse_args(argv)
+    verbose = arguments.v
+    set_loglevel(verbose=verbose)    # values [0, 1, 2] map to logging levels [WARNING, INFO, DEBUG]
+
+    # Step 0: Argument validation
+    # ---------------------------
+    # Load input and output filenames
+    fname_labels = arguments.l
+    fname_segmentation = arguments.i
+    fname_vertfile = arguments.vertfile
+    distance = arguments.distance
+    extent = arguments.extent
+    if arguments.o is not None:
+        fname_out = arguments.o
+    else:
+        path, file_name, ext = extract_fname(get_absolute_path(arguments.i))
+        fname_out = os.path.join(path, file_name + '_compression_metrics' + '.csv')
+
+    # Step 1. Load subject input files and get csv files (label image, segmentation)
+    # -----------------------------------------------------------
+    img_seg = Image(fname_segmentation).change_orientation('RPI')
+    img_labels = Image(fname_labels).change_orientation('RPI')
+    img_vertfile = Image(fname_vertfile).change_orientation('RPI')
+    # Check if segmentation, compression labels, and vertebral label files have same dimensions
+    if not img_seg.data.shape == img_labels.data.shape == img_vertfile.data.shape:
+        raise ValueError(f"Shape mismatch between compression labels [{img_labels.data.shape}], vertebral labels [{img_vertfile.data.shape}]"
+                         f" and segmentation [{img_seg.data.shape}]). "
+                         f"Please verify that your compression labels and vertebral labels were done in the same space as your input segmentation.")
```

Comment:
I think it might be good to move this whole section (input loading and shape validation) into the "Step 0: Argument validation section"